packages/preferences/src/browser/views/preference-searchbar-widget.tsx

```diff
@@ -53,19 +55,49 @@ export class PreferencesSearchbarWidget extends ReactWidget {
         }
     }
 
+    protected clearSearchResults = (e: React.MouseEvent): void => {
+        const search = document.getElementById('preference-searchbar') as HTMLInputElement;
+        if (search) {
+            search.value = '';
+            this.search(search.value);
+            this.update();
+        }
+    };
+
+    protected renderOptionContainer(): React.ReactNode {
+        const clearAllOption = this.renderOptionElement();
+        const resultsCount = this.renderResultsCount();
+        return <div className="option-buttons"> {this.searchbarRef.current?.value !== '' ? resultsCount : ''} {clearAllOption} </div>;
+    }
+
+    protected renderResultsCount(): React.ReactNode {
+        return <span
+            className="results-found"
+            title={`${this.searchCount} Results Found`}> {this.searchCount} Results Found </span>;
+    }
+
+    protected renderOptionElement(): React.ReactNode {
+        return <span
+            className={`clear-all option ${(this.searchbarRef.current?.value !== '') ? 'enabled' : ''}`}
+            title="Clear Search Results"
+            onClick={e => this.clearSearchResults(e)}
+        />;
+    }
+
     render(): React.ReactNode {
+        const optionContainer = this.renderOptionContainer();
         return (
-            <div className='settings-header'>
-                <div className="settings-search-container">
-                    <input
-                        type="text"
-                        placeholder="Search Settings"
-                        className="settings-search-input theia-input"
-                        onChange={this.handleSearch}
-                        ref={this.searchbarRef}
-                    />
-                </div>
-            </div >
+            <div className="settings-search-container">
+                <input
+                    type="text"
+                    id="preference-searchbar"
```

Comment:
We can have a `static` variable for the id which can be referenced (for example in `clearSearchResults`:

```ts
static readonly ID = 'preference-searchhbar'
```
```ts
protected clearSearchResults = (e: React.MouseEvent): void => {
  const search = document.getElementById(PreferencesSearchbarWidget.ID) as HTMLInputElement;
  ...
```